google/data_source_google_compute_vpn_gateway.go

```diff
@@ -0,0 +1,84 @@
+package google
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"google.golang.org/api/compute/v1"
+	"google.golang.org/api/googleapi"
+)
+
+func dataSourceGoogleComputeVpnGateway() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceGoogleComputeVpnGatewayRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"description": &schema.Schema{
```

Comment:
We order these fields "Required, Optional, Computed" - A field which is both Optional and Computed counts as Optional for this ordering.